File: data-structures/two-dimensional-arrays/visual-basic/all_words.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program


Module program

    ' The Main method is the default entry point for all VB programs
    Sub Main()
        Dim spellingWords(2, 2) As String ' Declare a two dimensional array

        spellingWords(0, 0) = "school"
        spellingWords(0, 1) = "pull"
        spellingWords(0, 2) = "where"
        spellingWords(1, 0) = "path"
        spellingWords(1, 1) = "floor"
        spellingWords(1, 2) = "sugar"
        spellingWords(2, 0) = "accident"
        spellingWords(2, 1) = "answer"
        spellingWords(2, 2) = "eight"

        For level = 0 To 2
            For word = 0 To 2
                Console.WriteLine(spellingWords(level, word))
            Next
        Next

    End Sub


End Module

File: data-structures/two-dimensional-arrays/python/all_words.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0

def main():
    spelling_words = [
        ["school", "pull", "where"],
        ["path", "floor", "sugar"],
        ["accident", "answer", "eight"]]
  
    for level in range(3):
        for word in range(3):
            print(spelling_words[level][word])


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()
